### PR TITLE
Rename "adoption" to "match"

### LINKED ISSUE #73: Rename `Adoption` model to `Match`
name change from `adoptions` to `matches` in db tables, it's basically a join table between pet and adopter

### PR DESCRIPTION
**Pull request by nsiwnf**
### Describe your change in plain English.

Renaming "Adoption" to "Match" since the described relationship could be adoption or fostering etc. 

### Link to the issue

Resolves #73

### CODE REVIEW

**Review by albertchae — APPROVED** [review]
Looks good in terms of renaming things, but just wanted to make sure we know what we're getting out of doing this. Having a common way to link pets to humans seems good but it also seems like fosters might have some different properties from adoptions (e.g. a pet could probably have multiple fosters but maybe one adoption, although I guess an adoption can be revoked and then adopted later? A foster probably has some desired duration whereas adoptions should be assumed permanent?)

Another point I'm thinking about is the Domain Driven Design principle of [ubiquitous language](https://martinfowler.com/bliki/UbiquitousLanguage.html) where we can run into "translation" problems if the technical term diverges from language used by domain experts https://thedomaindrivendesign.io/developing-the-ubiquitous-language/. 

Anyway just some thoughts, I don't think they block this change so approving the PR.

File: config/routes.rb [hunk 1]
```diff
@@ -30,8 +30,8 @@
   post "create_my_application", to: "adopter_applications#create"
   patch "my_application", to: "adopter_applications#update"
 
-  post "create_adoption", to: "adoptions#create"
-  delete "revoke_adoption", to: "adoptions#delete"
+  post "create_adoption", to: "matches#create"
+  delete "revoke_adoption", to: "matches#delete"
```
[the remark this answers is not in the document]
Comment (nsiwnf) [reply config/routes.rb:34]: Yeah, I wasn't sure how much to rename - baja pet rescue is adoption-based and I figured we can address these details once we do introduce a concept of "foster"?

[the remark this answers is not in the document]
Comment (albertchae) [reply config/routes.rb:34]: yeah makes sense!
